docs/resources/alks_iamrole.md

```diff
@@ -92,7 +92,8 @@ The following arguments are supported:
 
 * `name` - (Optional/Computed) The name of the ALKS IAM role which will be reflected in AWS and the ALKS UI.
 * `name_prefix` - (Optional/Computed) A prefix for a generated name of the ALKS IAM role which will be reflected in AWS and the ALKS UI.
-* `type` - (Required) The role type to use. To see a list of available roles, [call this endpoint](https://pages.ghe.coxautoinc.com/ETS-CloudAutomation/ALKS-Documentation/#/aws-role-type-rest-service/getAllAwsRoleTypesUsingGET).
+* `type` - (Required) The role type to use.  This argument is mutually exclusive with `assume_role_policy`. To see a list of available roles, [call this endpoint](https://pages.ghe.coxautoinc.com/ETS-CloudAutomation/ALKS-Documentation/#/aws-role-type-rest-service/getAllAwsRoleTypesUsingGET).
+* `assume_role_policy` - (Required) A JSON string representing the trust policy for the role. This is only supported for single-service trust policies trusting an approved AWS service.  This argument is mutually exclusive with `type`.
```

Comment:
Referencing documentation [examples](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lambda_function#filename) from the AWS provider, XORd attributes are Optional with a note: "Exactly one of this, this, or this must be specified". I think wording it this way is clearer.

Also, This page can use some serious upgrades and edits. For instance, the introduction summary is grammatically incorrect. Also The details regarding custom trust policy usage starting on line 32 have changed. The guidelines should be a translation of [this](https://ghe.coxautoinc.com/ETS-CloudAutomation/Fides/blob/dc63aca7bde13eaa2729008f6724712e91197089/bundles/main.rego#L425) function in Fides. 

And Lastly, the URL provided in the `type` attr description is incorrect. It should be the following: https://pages.ghe.coxautoinc.com/ETS-CloudAutomation/ALKS-Documentation/#operation/getAllAwsRoleTypes

All or some of these concerns can be address in this PR. Anything that seems like it falls out of scope can be punted to a new story.

Reply:
Thanks for the catch.   I updated the text to use the "Exactly one of this, this, or this must be specified" pattern.  Also updated the URL to the correct link.

For your other concerns, I will make a separate defect.